File: CS/CustomTreeList-NodeHeight/MyTreeList.cs
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.Nodes;

namespace CustomTreeList_NodeHeight {
    public class MyTreeList : TreeList {
        public MyTreeList() : base() {
            OptionsBehavior.AutoNodeHeight = false;
        }

        protected override TreeListNode CreateNode(int nodeID, TreeListNodes owner, object tag) {
            return new MyTreeListNode(nodeID, owner);
        }

        protected override void InternalNodeChanged(TreeListNode node, TreeListNodes nodes, NodeChangeTypeEnum changeType) {
            if (changeType == NodeChangeTypeEnum.User1)
                LayoutChanged();
            base.InternalNodeChanged(node, nodes, changeType);
        }
        protected override void RaiseCalcNodeHeight(TreeListNode node, ref int nodeHeight) {
            MyTreeListNode myNode = node as MyTreeListNode;
            if (myNode != null)
                nodeHeight = myNode.Height;
            else
                base.RaiseCalcNodeHeight(node, ref nodeHeight);
        }
        public virtual int DefaultNodesHeight { get { return 18; } }
    }

    public class MyTreeListNode : TreeListNode {
        const int minHeight = 5;
        int height;
        public MyTreeListNode(int id, TreeListNodes owner) : base(id, owner) {
            this.height = (owner.TreeList as MyTreeList).DefaultNodesHeight;
        }
        public int Height {
            get { return height; }
            set {
                if (Height == value || value < minHeight) return;
                height = value;
                Changed(NodeChangeTypeEnum.User1);
            }
        }
    }
}

File: VB/CustomTreeList-NodeHeight/MyTreeList.vb
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.Nodes

Namespace CustomTreeList_NodeHeight
    Public Class MyTreeList
        Inherits TreeList

        Public Sub New()
            MyBase.New()
            OptionsBehavior.AutoNodeHeight = False
        End Sub

        Protected Overrides Function CreateNode(ByVal nodeID As Integer, ByVal owner As TreeListNodes, ByVal tag As Object) As TreeListNode
            Return New MyTreeListNode(nodeID, owner)
        End Function

        Protected Overrides Sub InternalNodeChanged(ByVal node As TreeListNode, ByVal nodes As TreeListNodes, ByVal changeType As NodeChangeTypeEnum)
            If changeType Is NodeChangeTypeEnum.User1 Then
                LayoutChanged()
            End If
            MyBase.InternalNodeChanged(node, nodes, changeType)
        End Sub
        Protected Overrides Sub RaiseCalcNodeHeight(ByVal node As TreeListNode, ByRef nodeHeight As Integer)
            Dim myNode As MyTreeListNode = TryCast(node, MyTreeListNode)
            If myNode IsNot Nothing Then
                nodeHeight = myNode.Height
            Else
                MyBase.RaiseCalcNodeHeight(node, nodeHeight)
            End If
        End Sub
        Public Overridable ReadOnly Property DefaultNodesHeight() As Integer
            Get
                Return 18
            End Get
        End Property
    End Class

    Public Class MyTreeListNode
        Inherits TreeListNode

        Private Const minHeight As Integer = 5

        Private height_Renamed As Integer
        Public Sub New(ByVal id As Integer, ByVal owner As TreeListNodes)
            MyBase.New(id, owner)
            Me.height_Renamed = (TryCast(owner.TreeList, MyTreeList)).DefaultNodesHeight
        End Sub
        Public Property Height() As Integer
            Get
                Return height_Renamed
            End Get
            Set(ByVal value As Integer)
                If Height = value OrElse value < minHeight Then
                    Return
                End If
                height_Renamed = value
                Changed(NodeChangeTypeEnum.User1)
            End Set
        End Property
    End Class
End Namespace


